src/com/ppm/integration/agilesdk/connector/octane/OctaneUserIntegration.java

```diff
@@ -89,25 +92,43 @@
     }
 
     /**
-     * TODO need classify users to add security and product.
      * @param user
+     * @param roles contain user role information.
      * @param agileProjectJson
      *            {"securityGroups":{"admin":["securityGroupReferenceCode1"],"users":["securityGroupReferenceCode2"]},"productLicenses":{"admin":[productId1],"users":[productId2]},"agileProjectValue":{"WORKSPACE_ID":1003,"SHARED_SPACE_ID":1003}}
      */
-    private void addSecurityGroupAndLicenseToUsers(AgileDataUser user, JSONObject agileProjectJson) {
-        // security and license
-        JSONObject securityGroupsJson = agileProjectJson.getJSONObject("securityGroups");
-        JSONArray adminSecurityGroups = securityGroupsJson.getJSONArray("admin");
-        JSONArray userSecurityGroups = securityGroupsJson.getJSONArray("users");
-        JSONObject productLicensesJson = agileProjectJson.getJSONObject("productLicenses");
-        JSONArray adminLicenses = productLicensesJson.getJSONArray("admin");
-        JSONArray userLicenses = productLicensesJson.getJSONArray("users");
-
-        List<Long> userLicensesList = JSONArray.toList(userLicenses, Long.class);
-        List<String> userSecurityList = JSONArray.toList(userSecurityGroups, String.class);
-
-        user.setSecurityGroupCodes(userSecurityList);
-        user.setProductIds(userLicensesList);
+    private void addSecurityGroupAndLicenseToUsers(AgileDataUser user, JSONObject roles, JSONObject agileProjectJson) {
+        if (!roles.isNullObject()) {
+            JSONArray rolesArray = roles.getJSONArray("data");
+            List<String> roleList = new ArrayList<>();
```

Comment:
I think that instead of this long manual processing of JSonObject, you could use GSon (or another library) and parse it into a Java Object directly.